src/jit/lower.cpp

```diff
@@ -372,17 +372,16 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
         }
         break;
     case GT_CALL:
-        NYI("Call with TYP_LONG return value");
         break;
     case GT_RETURN:
-        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG);
+        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG || tree->gtOp.gtOp1->OperGet() == GT_CALL);
```

Comment:
In codegenxarch.cpp, I don't see logic to handle GT_RETURN(GT_CALL) case.  genReturn() is noway asserting that it is always a GT_LONG.  For x86, we need to add logic on the lines similar to genStructReturn().